### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by mschneid**
Just as a proposal. I don't know your mindset about Java 8 et cetera...

### CODE REVIEW

**Comment by mkouba** [comment]
That's a lot of commits :-) And I like most of the changes. Note that first versions of trimou supported JDK6 ;-)

**Comment by mschneid** [comment]
@mkouba I know that Trimou startd with JDK 6 and it was my last intention to criticize anything about your project. I like the approach you have chosen for a simple and nice template engine. I just begun to play with it and hope I can help you to make it even better.

**Comment by mkouba** [comment]
@mschneid I did not mean it that way... I really appreciate your intetest and your contributions are welcome!

File: core/src/main/java/org/trimou/util/Arrays.java [hunk 1]
```diff
@@ -37,9 +37,9 @@ public static boolean contains(Object[] array, Object object) {
         if (array == null) {
             return false;
         }
-        for (int i = 0; i < array.length; i++) {
-            if ((object != null && object.equals(array[i]))
-                    || (object == null && array[i] == null)) {
+        for (final Object anArray : array) {
```
Comment (mkouba) [inline core/src/main/java/org/trimou/util/Arrays.java]: `anArray` should be array `element`.

File: extensions/cdi/src/main/java/org/trimou/cdi/resolver/CDIBeanResolver.java [hunk 2]
```diff
@@ -107,14 +107,9 @@ public Object resolve(Object contextObject, String name,
...
-        Optional<Bean<?>> bean = beanCache.get(name);
-
-        if (!bean.isPresent()) {
-            // Unsuccessful lookup already performed
-            return null;
-        }
-        return getReference(bean.get(), context);
+        return beanCache.get(name)
+                .map(bean -> getReference(bean, context))
+                .orElse(null);
```
Comment (mkouba) [inline extensions/cdi/src/main/java/org/trimou/cdi/resolver/CDIBeanResolver.java]: I find the original code more readable and more performant.

File: core/src/main/java/org/trimou/engine/config/SecurityActions.java [hunk 3]
```diff
@@ -67,12 +67,7 @@ static ClassLoader getClassLoader(final Class<?> clazz) {
             return clazz.getClassLoader();
         }
         return AccessController
-                .doPrivileged(new PrivilegedAction<ClassLoader>() {
-                    @Override
-                    public ClassLoader run() {
-                        return clazz.getClassLoader();
-                    }
-                });
+                .doPrivileged((PrivilegedAction<ClassLoader>) clazz::getClassLoader);
```
Comment (mkouba) [inline core/src/main/java/org/trimou/engine/config/SecurityActions.java]: I do not like the usage of method references for `PrivilegedAction` stuff... esp. the need for casting.

File: core/src/main/java/org/trimou/engine/config/SecurityActions.java [hunk 4]
```diff
@@ -33,12 +33,7 @@ static String getSystemProperty(final String key) {
         if (System.getSecurityManager() == null) {
             return System.getProperty(key);
         }
-        return AccessController.doPrivileged(new PrivilegedAction<String>() {
-            @Override
-            public String run() {
-                return System.getProperty(key);
-            }
-        });
+        return AccessController.doPrivileged((PrivilegedAction<String>) () -> System.getProperty(key));
```
Comment (mkouba) [inline core/src/main/java/org/trimou/engine/config/SecurityActions.java]: I wouldn't use lambdas here (`PrivilegedAction`)...

**Pull request merged**